node/node.go

```diff
@@ -270,6 +269,7 @@ func newHeaderValidator(logger log.Logger) p2p.GossipValidator {
 			logger.Error("failed to validate header", "error", err)
 			return false
 		}
+		incomingHeaderCh <- headerMsg
```

Comment:
One of the key points in #126 is to remove double-deserialization. `incommingHeaderCh` type should be changed to `chan *types.Header`, and `&header` should be send, instead of re-sending `GossipMessage`.

Reply:
Seems like I don't understand the whole flow. Isn't the header deserialized only once in `headerReadLoop`  https://github.com/celestiaorg/optimint/blob/39ffb65790ea320217708a5eff734568b4dffceb/node/node.go#L144?